SU2_CFD/src/numerics/species/species_sources.cpp

```diff
@@ -58,18 +58,86 @@ CSourceAxisymmetric_Species<T>::CSourceAxisymmetric_Species(unsigned short val_n
                                                          const CConfig* config)
     : CSourceBase_Species(val_nDim, val_nVar, config),
     idx(val_nDim, config->GetnSpecies()) {
-  implicit = (config->GetKind_TimeIntScheme_Flow() == EULER_IMPLICIT);
+  implicit = (config->GetKind_TimeIntScheme_Species() == EULER_IMPLICIT);
+  viscous = config->GetViscous();
+  inc_rans = (config->GetKind_Solver() == MAIN_SOLVER::INC_RANS) || (config->GetKind_Solver() == MAIN_SOLVER::DISC_ADJ_INC_RANS);
+  incompressible = (config->GetKind_Regime() == ENUM_REGIME::INCOMPRESSIBLE);
+
+  Sc_t = config->GetSchmidt_Number_Turbulent();
+
 }
 
 template <class T>
 CNumerics::ResidualType<> CSourceAxisymmetric_Species<T>::ComputeResidual(const CConfig* config) {
-  for (unsigned short iVar = 0; iVar < nVar; iVar++) residual[iVar] = 0.0;
+  
+  su2double yinv,Density_i,Velocity_i[3];
 
-  if (implicit) {
-    for (unsigned short iVar = 0; iVar < nVar; iVar++) {
-      for (unsigned short jVar = 0; jVar < nVar; jVar++) jacobian[iVar][jVar] = 0.0;
+  /*--- Preaccumulation ---*/
+  AD::StartPreacc();
+  AD::SetPreaccIn(ScalarVar_i, nVar);
+  AD::SetPreaccIn(Volume); 
+
+  if (incompressible) {
+    AD::SetPreaccIn(V_i, nDim+6);
+  }
+  else {
+    AD::SetPreaccIn(V_i, nDim+7);
+  }
+
+  /*--- Initialization. ---*/
+  for (auto i_var = 0; i_var < nVar; i_var++) {
+    residual[i_var] = 0.0;
+    for (auto j_var = 0; j_var < nVar; j_var++) {
+      jacobian[i_var][j_var] = 0.0;
+    }
+  }
+
+  /*--- Contribution due to 2D axisymmetric formulation ---*/
+  if (Coord_i[1] > EPS) {
+
+    AD::SetPreaccIn(Coord_i[1]);
+    AD::SetPreaccIn(Diffusion_Coeff_i, nVar);
+    AD::SetPreaccIn(ScalarVar_Grad_i, nVar, nDim);      
+
+    yinv = 1.0/Coord_i[1];
+
+    /*--- The incompressible density. Note that this is different for compressible flows ---*/
+
+    Density_i = V_i[nDim+2];
+
+    /*--- Set primitive variables at points iPoint. ---*/
+    
+    for (auto iDim = 0u; iDim < nDim; iDim++)
+      Velocity_i[iDim] = V_i[iDim+1];
+
+    /*--- Inviscid component of the source term. ---*/
+    
+    for (auto iVar=0u; iVar < nVar; iVar++)
+      residual[iVar] -= yinv*Volume*Density_i*ScalarVar_i[iVar]*Velocity_i[1];
+
+    if (implicit) {
+      for (auto iVar=0u; iVar < nVar; iVar++) {
+        jacobian[iVar][iVar] -= yinv*Volume*Velocity_i[1];
+      }
+    }
+
+    /*--- Add the viscous terms if necessary. ---*/
+    
+    if (config->GetViscous()) {
+      Eddy_Viscosity_i       = V_i[nDim+5];
+
+      su2double Mass_Diffusivity_Tur = 0.0;
+      if (inc_rans)
+        Mass_Diffusivity_Tur = Eddy_Viscosity_i/Sc_t;
```

Comment:
What if the diffusivity model is not constant Schmidt?

Reply:
This is the turbulent Schmidt number, Sc_t for species is like turbulent Prandtl number Pr_t for flow.

@Cristopher-Morales I see that in the CFluidModel::MakeMassDiffusivityModel the turbulent Schmidt is only active when the laminar Schmidt number is used. I think we should also use Sc_t for the other 2 diffusion models, since the turbulent diffusion is independent of the laminar part. 
By the way, the diffusion is $$\frac{\mu}{Sc} + \frac{\mu_t}{Sc_t}$$ and it looks like the laminar part is wrong in CConstantSchmidtRans.

Reply:
Hi, Yes I noticed that. I think I made a mistake in the CConstantSchmidtRans. I think we could remove the CConstantSchmidtRans and keep the implementation as it is. Because the $D_{lam}$ is the one that can be: CONSTANT,CONSTANT_SCHIMDT or UNITY_LEWIS. For the turbulent part, we have only the turbulent Schmidt. Other option can be to follow a similar implementation as we do for the thermal conductivity which has the option of TURBULENT_PRANDTL. But I think that would requiere more job because in that case the diffusivity would be laminar or laminar+turbulent inmediatly after the MassDiffusivityPointers are created. Please let me knwo what you think and my apologizes for my mistake in that implementation.

Reply:
Ok, if I understand correctly, CConstantSchmidtRans is not used, and the correct laminar and turbulent Schmidt number is used in SU2 in species_diffusion.hpp (in residualcalc).